.github/actions/BuildCirct/action.yml

```diff
@@ -4,39 +4,25 @@ description: "Builds CIRCT, which is used for the HLS backend"
 runs:
   using: "composite"
   steps:
-    # Clone jlm-eval-suite
-    - name: Clone jlm-test-suite
-      run: git clone https://github.com/phate/jlm-eval-suite.git
-      shell: bash
-
-    # Extract the CIRCT submodule hash for use in the cache key
-    - name: Get CIRCT hash
+    - name: "Get the commit used for building CIRCT and use it as the cache key"
       id: get-circt-hash
       run: |
-        cd jlm-eval-suite
-        echo "hash=$(git rev-parse @:./circt)" >> $GITHUB_OUTPUT
+        echo "hash=$(./circt.sh --get-commit-hash)" >> $GITHUB_OUTPUT
       shell: bash
 
-    # Try to fetch CIRCT from the cache
-    - name: Cache CIRCT
+    - name: "Try to fetch CIRCT from the cache"
       id: cache-circt
       uses: actions/cache@v3
       with:
         path: |
-          jlm-eval-suite/circt/local
+          ./build-circt/circt
         key: ${{ runner.os }}-circt-${{ steps.get-circt-hash.outputs.hash }}
 
-    # Install dependencies if we didn't hit in the cache
     - name: "Install LLVM, Clang, MLIR, and Ninja"
-      if: steps.cache-circt.outputs.cache-hit != 'true'
       uses: ./.github/actions/InstallLlvmDependencies
```

Comment:
Do we need to install LLVM dependencies if we do not need to build anything?

Reply:
Your comment for the MLIR frontend was that I had LLVM followed by MLIR in the actual tests, and that the MLIR build action depended implicitly on LLVM.
This is the same case. There is an implicit dependency, but only if CIRCT Is being built. But the test building CIRCT needs LLVM.